### PR TITLE
new project with Sphinx theme

### PR DESCRIPTION
**Pull request by lauranovich**
new project 
add sphinx theme and a basic page

### CODE REVIEW

**Comment by tzach** [comment]
Nice start!

**Comment by tzach** [comment]
Next tasks:
- Deploy to https://scylladb.github.io/scylla-monitoring 
- Copy latest doc to the master branch

Once this is working, we can add more releases.

**Comment by lauranovich** [comment]
What elements do we want on the header?
how does it work with links?

**Comment by tzach** [comment]
> What elements do we want on the header?

See example here:
https://scylladb.github.io/python-driver/master/index.html
(replace https://scylladb.github.io/python-driver/ with https://scylladb.github.io/scylla-monitoring)

> how does it work with links?

Not sure I understand the question. 
See example here: https://github.com/scylladb/python-driver/blob/master/docs/conf.py#L103

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/docs/_build/
```
Comment (tzach) [inline .gitignore]: Please remove everything under _build and add the following to .gitignore:
docs/_build/
docs/venv/
*.swp
*.pyc
docs/bin/
docs/include/
docs/lib/
docs/lib64
docs/pip-selfcheck.json
docs/.Python

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[tool.poetry]
+name = "sphinx-docs"
```
Comment (tzach) [inline pyproject.toml]:
```suggestion
name = "scylla-monitoring-docs"
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+description = "ScyllaDB Documentation"
```
Comment (tzach) [inline pyproject.toml]:
```suggestion
description = "Scylla Monitoring Documentation"
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+version = "1.3"
```
Comment (tzach) [inline pyproject.toml]:
```suggestion
version = "0.1"
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+authors = ["ScyllaDB Documentation Contributors"]
+
+[tool.poetry.dependencies]
+python = "^3.7"
+pyyaml = "^5.3"
+pygments = "2.2.0"
+recommonmark = "0.5.0"
+sphinx-scylladb-theme = "0.1.6"
```
Comment (tzach) [inline pyproject.toml]:
```suggestion
sphinx-scylladb-theme = "0.1.8"
```

**Comment by tzach** [comment]
File `docs/_utils/__pycache__/cql.cpython-38.pyc` and all pyc files should not be part of Git

File: docs/_utils/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh -e
+
+ORIGIN="https://github.com/scylladb/scylla-docs.git"
```
Comment (tzach) [inline docs/_utils/deploy.sh]:
```suggestion
ORIGIN="https://github.com/scylladb/scylla-monitoring.git"
```

File: docs/_utils/redirections.yaml [hunk 7]
```diff
@@ -0,0 +1,258 @@
+### a dictionary of redirections
```
Comment (tzach) [inline docs/_utils/redirections.yaml]: This file is not relevant for this project

File: docs/index.rst [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+* `Scylla Monitor Support Matrix <matrix>`_
+* `Install Scylla Monitor Stack <monitoring_stack>`_
+* :doc:`The start-all.sh script <start_all>`
+* `Scylla Monitor Interfaces <monitoring_apis>`_
+* `Deploy Scylla Monitor Without Docker <monitor_without_docker>`_
+* `Troubleshoot Scylla Monitor <monitor_troubleshoot>`_
+* :doc:`CQL Optimization Dashboard <cql_optimization>`
+* :doc:`Adding and Modifying Dashboards <updating_dashboard>`
+* :doc:`Alerting <alerting>`
```
Comment (tzach) [inline docs/index.rst:43]: The original page also include:
```
**Additional Information**:

* `Scylla Monitoring Support Matrix <matrix>`_
* `Upgrade Scylla Monitoring </upgrade/upgrade-monitor/>`_
* `Monitoring lesson <https://university.scylladb.com/courses/scylla-operations/lessons/admin-procedures-and-basic-monitoring/>`_ on Scylla University

.. image:: monitor.png
```

File: docs/index.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+   Troubleshoot Monitor <monitor_troubleshoot>
+
+.. include:: /_common/monitor-description.rst
+
+The Scylla Monitor Stack consists of three components, wrapped in Docker containers:
+
+* `prometheus` - collects and stores metrics
+* `alertmanager` - handles alerts
+* `grafana` - dashboard server
+
```
Comment (tzach) [inline docs/index.rst:29]: For older version of Scylla Monitoring see `here <https://docs.scylladb.com/operating-scylla/monitoring/>`_

File: docs/monitor_troubleshoot.rst [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+
+
+.. note::
+
+   Avoid running Docker containers as root.
+
+The preferred way of running the container is using a non root user.
+See the `monitoring`_ Docker post-installation section.
+
+.. _`monitoring`: /operating-scylla/monitoring/3.4/monitoring_stack/#docker-post-installation
```
Comment (tzach) [inline docs/monitor_troubleshoot.rst]:
```suggestion
.. _`monitoring`: /monitoring_stack/#docker-post-installation 
```

Need to check
All link should be update to this project and not include version

File: docs/monitor_without_docker.rst [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+.. include:: min-prod-hw.rst
+
+The main item to set an alert on is the available disk space in the monitoring system. Data is indefinitely accrued on the Prometheus data directory. The current monitoring solution does not churn data.
+
+Install Scylla Monitor
+----------------------
+
+The following procedure uses a ``CentOS 7`` based instance
+
+1. Download the latest Scylla Monitoring release. Currently it is |mon_version|.
```
Comment (tzach) [inline docs/monitor_without_docker.rst]:
```suggestion
1. Download the latest Scylla Monitoring release.
```

Remove all version info and |mon_version| version

File: docs/_utils/redirections.yaml (hunk 7 above)
Comment (lauranovich) [reply docs/_utils/redirections.yaml]: ok I'll remove it

**Comment by lauranovich** [comment]
@tzach - Hope I didn't mess this one up. Please review

**Comment by tzach** [comment]
I'm getting the following error:
```
Because scylla-monitoring-docs depends on sphinx-scylladb-theme (0.1.8) which doesn't match any versions, version solving failed.
```

But there is a 0.1.8 version https://pypi.org/project/sphinx-scylladb-theme/

@dgarcia360 any idea?

**Comment by lauranovich** [comment]
@tzach I have the same errror message
@dgarcia360 - please let me know what to do

**Comment by dgarcia360** [comment]
@tzach @lauranovich

Before the error, you will see the following warning ``The lock file is not up to date with the latest changes in pyproject.toml.`` This appears when the ``pyproject.yml`` file has been edited manually but the ``poetry.lock`` file has not been updated accordingly.

To solve the error, you can either:

a) (Recommended) Run ``poetry update`` in the root project folder.
b) Delete the file ``poetry.lock`` and run ``make preview`` again.

**Extra info about how poetry works**

I'd like to explain to you why both files should match, it could help to understand better how poetry works and solve other problems in the future. As you have seen, Poetry works mainly with two files:

-  ``pyproject.yml``:  Describes the minimum/maximum/exact version of each dependency to be installed. Example: sphinx-scylladb-theme<0.1.6.
- ``poetry.lock``: Describes the exact version being used to avoid installing modules from the same ``pyproject.yml`` result in two different installs. This is auto-generated from ``pyproject.yml``. Example: sphinx-scylladb-theme==0.1.5, which is < 0.1.6.

The current [pyproject.yml](https://github.com/scylladb/scylla-monitoring/pull/984/files#diff-522adf759addbd3b193c74ca85243f7dR12) file currently points to sphinx-scylladb-theme==**0.1.8** but the dependency in [poetry.lock](https://github.com/scylladb/scylla-monitoring/pull/984/files#diff-41fe8bebc1a2a52eb5321b759e40b3a8R748) still is **0.1.6**, which is different to 0.1.8. When we run ``poetry update``, the file ``poetry.lock`` will be regenerated according to the latest ``pyproject.yml`` specification.

**Comment by tzach** [comment]
@amnonh this looks good to me.
I think we can merge, there is no, or very little risk

**Comment by lauranovich** [comment]
I am correcting an error on my end, I will need to send a new commit - wait before you merge please

**Comment by lauranovich** [comment]
@tzach @amnonh  - all set - can we merge?

**Comment by amnonh** [comment]
I've tested it locally and saw that it works for me

**Pull request merged**